### PR TITLE
Move plugin version definitions to properties.

### PR DESCRIPTION
**Pull request by tswast**
This a workaround for a missing feature in the codehaus versions maven
plugin where is cannot update plugins directly, but it can detect when
there are new plugins and update the related properties. https://github.com/mojohaus/versions-maven-plugin/issues/8#issuecomment-225593578

I don't actually update anything (or I tried not to, anyway). I'll get dpebot to do that.

@lesv

### CODE REVIEW

**Comment by tswast** [comment]
@dpebot please merge when green

File: storage/xml-api/serviceaccount-appengine-sample/pom.xml [hunk 1]
```diff
@@ -17,10 +31,15 @@
   <packaging>war</packaging>
 
   <properties>
+    <maven.compiler.target>1.6</maven.compiler.target>
+    <maven.compiler.source>1.6</maven.compiler.source>
```
Comment (lesv) [inline storage/xml-api/serviceaccount-appengine-sample/pom.xml]: 1.6 -- really?

File: vision/face-detection/pom.xml [hunk 2]
```diff
@@ -29,6 +29,11 @@
     <relativePath>../..</relativePath>
   </parent>
 
+  <properties>
+    <maven.compiler.target>1.7</maven.compiler.target>
```
Comment (lesv) [inline vision/face-detection/pom.xml]: Probably should be 1.8

File: vision/label/pom.xml [hunk 3]
```diff
@@ -29,6 +29,11 @@
     <relativePath>../..</relativePath>
   </parent>
 
+  <properties>
+    <maven.compiler.target>1.7</maven.compiler.target>
```
Comment (lesv) [inline vision/label/pom.xml]: Probably should be 1.8

File: vision/landmark-detection/pom.xml [hunk 4]
```diff
@@ -29,6 +29,11 @@
     <relativePath>../..</relativePath>
   </parent>
 
+  <properties>
+    <maven.compiler.target>1.7</maven.compiler.target>
```
Comment (lesv) [inline vision/landmark-detection/pom.xml]: Probably should be 1.8

**Comment by lesv** [comment]
I'm generally happy with these changes, a few nits that can be ignored.

File: storage/xml-api/serviceaccount-appengine-sample/pom.xml (hunk 1 above)
Comment (tswast) [inline storage/xml-api/serviceaccount-appengine-sample/pom.xml]: /shruggie That's what it was. I was a bit surprised, too. I'll update that one, too.

**Pull request merged**